apis/v1beta1/config.go

```diff
@@ -88,11 +88,24 @@ func (c *AnyConfig) UnmarshalJSON(b []byte) error {
 }
 
 // MarshalJSON specifies how to convert this object into JSON.
-func (c *AnyConfig) MarshalJSON() ([]byte, error) {
-	if c == nil {
+func (c AnyConfig) MarshalJSON() ([]byte, error) {
+	if c.Object == nil {
 		return []byte("{}"), nil
 	}
-	return json.Marshal(c.Object)
+
+	nonNilMap := make(map[string]interface{})
+
+	for k, v := range c.Object {
+		if v == nil {
+			nonNilMap[k] = nil
+		} else if emptyMap, ok := v.(map[string]interface{}); ok && len(emptyMap) == 0 {
+			nonNilMap[k] = emptyMap
+		} else {
+			nonNilMap[k] = v
+		}
+	}
```

Comment:
It seems to me that this just makes a shallow copy of `c.Object`. Does that actually change anything?